src/main/java/net/atos/zac/zoeken/ZoekenService.java

```diff
@@ -80,22 +86,29 @@ public ZoekResultaat<ZaakZoekObject> zoek(final ZoekParameters zoekZaakParameter
             }
         });
 
-        zoekZaakParameters.getFilterQueries().forEach((veld, waarde) -> query.addFilterQuery(String.format("%s:\"%s\"", veld, waarde)));
+        zoekParameters.getFilterQueries().forEach((veld, waarde) -> query.addFilterQuery(String.format("%s:\"%s\"", veld, waarde)));
 
         query.setFacetMinCount(1);
         query.setFacetMissing(true);
+        query.setFacet(true);
         query.setParam("q.op", SimpleParams.AND_OPERATOR);
-        query.setRows(zoekZaakParameters.getRows());
-        query.setStart(zoekZaakParameters.getStart());
-        query.addSort(zoekZaakParameters.getSortering().sorteerVeld().getVeld(),
-                      zoekZaakParameters.getSortering().richting() == SorteerRichting.DESCENDING ? SolrQuery.ORDER.desc : SolrQuery.ORDER.asc);
-        if (zoekZaakParameters.getSortering().sorteerVeld() != SorteerVeld.IDENTIFICATIE) {
+        query.setRows(zoekParameters.getRows());
+        query.setStart(zoekParameters.getStart());
+        query.addSort(zoekParameters.getSortering().sorteerVeld().getVeld(),
+                      zoekParameters.getSortering().richting() == SorteerRichting.DESCENDING ? SolrQuery.ORDER.desc : SolrQuery.ORDER.asc);
+        if (zoekParameters.getSortering().sorteerVeld() != SorteerVeld.IDENTIFICATIE) {
             query.addSort(SorteerVeld.IDENTIFICATIE.getVeld(), SolrQuery.ORDER.desc);
         }
         try {
             final QueryResponse response = solrClient.query(query);
-            final ZoekResultaat<ZaakZoekObject> zoekResultaat = new ZoekResultaat<>(response.getBeans(ZaakZoekObject.class),
-                                                                                    response.getResults().getNumFound());
+            final List<ZoekObject> zoekObjecten = new ArrayList<>(response.getResults().size());
+            response.getResults().forEach(solrDocument -> {
+                final ZoekObjectType zoekObjectType = ZoekObjectType.valueOf((String) solrDocument.get("type"));
+                final ZoekObject bean = solrClient.getBinder().getBean(zoekObjectType.getZoekObjectClass(), solrDocument);
```

Comment:
verander de forEach in een map() en beeindig de stream met een toList. dan heb je meteen een list van zoekObjecten